src/properties.jl

```diff
@@ -144,6 +148,10 @@ real(φ::EuclideanVector) = EuclideanVector(real(φ.val))
 complex(φ::EuclideanVector) = EuclideanVector(complex(φ.val))
 complex(::Type{EuclideanVector{T}}) where {T} = EuclideanVector{complex(T)}
 
+# TODO MATTHIAS - IS THIS RIGHT, OR SHOULD IT DEPEND ON T? 
+isrealB(::EuclideanVector) = true 
+isrealAA(::EuclideanVector) = false 
+
```

Comment:
@MatthiasSachs Please check that you agree

Reply:
I would think that it should depend on the in isrealB, i.e., I think we should have

isrealB(::EuclideanVector{<: Real}) = true 
isrealB(::EuclideanVector{<: Complex}) = false 
isrealAA(::EuclideanVector) = false

I will try to test this myself tomorrow!

Reply:
Ok, but only if having complex covariant vectors makes sense at all? Does it? Or should Euclidean covariant vectors just be restricted to be real?

Reply:
(The fewer code-paths the fewer bugs…)

Reply:
Anyhow if you want both real and complex variants then we also need tests for both

Reply:
> Ok, but only if having complex covariant vectors makes sense at all? Does it? Or should Euclidean covariant vectors just be restricted to be real?

yes, I am wondered about that myself. Do we have applications for that?

Reply:
I can’t think of any. If we want two vectors then we just get two vectors.//

Reply:
OK,  I think there is an issue: in the current implementation we have to initialize an euclidian vector by
φ = ACE.EuclideanVector(Complex{Float64})
irrespectively of whether in the end we want the basis functions itself to be real or complex. I think this is very confusing. I in order to resolve the I think we either need to require EuclidianVector to be always real (as in your current implementation) or we need to add an additional parameter to EuclidianVector which indicates whether the property is supposed to be treated as complex or real and on which we can dispatch. Does this make sense?

Reply:
Let’s require that it is always real, and change the type parameter to just Float64.

Reply:
Can tou have a go at this? Ping me if you run into problems

Reply:
OK, I have a working version ready on co/realsymm. Should I push directly to that branch?

Reply:
Yes please - also remove the comment with my q for you

Reply:
shouldn't there be other changes as well?

Reply:
ah-sorry, I missed the previous commit

Reply:
looks all got, just running the tests now